.github/workflows/release-branches-check.yml

```diff
@@ -0,0 +1,53 @@
+name: Release branches check
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - juan-fernandez/add-job-to-check-release-branches
+
+jobs:
+  check-release-branch-v2:
+    if: "!contains(github.event.pull_request.labels.*.name, 'dont-land-on-v2.x')"
```

Comment:
my first attempt was using `strategy.matrix` to check `v2.x` and `v3.x`, instead of repeating so much of the job, but this `if` part (to take into account PR labels) seems to be hard (impossible?) with `strategy`

Reply:
You could always just make an action and use the action in both jobs.